### ISSUE TITLE
Strange gutter behavior (diff information not updating, line numbers overlapping)

### ISSUE THREAD

**Issue by lee-dohm** [atom/atom#6545]
I have a repro for the diff information not updating:

1. Ensure that Atom is your editor for command-line Git commands `core.editor = atom --wait`
1. Check out a git repository (I used a random Atom one)
1. Open a file in the git repository
1. Add a bunch of text (my mechanical keyboard sounds awesome when monkey typing in random characters :grinning:)
1. Save the file with <kbd>Cmd+S</kbd>
1. In a Terminal window, type `git commit -a`
1. Wait for the new Atom window show up
1. Type "DELETE THIS COMMIT" as the commit message
1. Save the file with <kbd>Cmd+S</kbd>
1. Close the Git Commit Atom window with <kbd>Shift+Cmd+W</kbd>
1. Switch back to the original Atom window with <kbd>Cmd+Tab</kbd>

**Expected:** Git diff information to disappear
**Actual:** Git diff information remains

![screen shot 2015-04-27 at 10 27 40 pm](https://cloud.githubusercontent.com/assets/1038121/7363350/f72b852e-ed2d-11e4-85a0-5f3877f432d7.png)

![screen shot 2015-04-27 at 10 28 25 pm](https://cloud.githubusercontent.com/assets/1038121/7363358/115fc392-ed2e-11e4-8264-946406f6f839.png)

Some things that got Atom to clear the diff info (but not reliably):

1. Keyboard tabbing to Slack and back
1. Closing the dev tools pane

**Comment by izuzak** [comment]
@lee-dohm :zap: thanks for the steps! 

@nathansobo @kevinsawicki This is probably the behavior we were trying to reproduce yesterday. I was able to reproduce the problem by following the steps above.

**Comment by nathansobo** [comment]
@izuzak Unfortunately I don't think this is quite the issue we were seeing. What we saw was the diff sticking around and not being cleared.

**Comment by izuzak** [comment]
@nathansobo I _think_ that's what @lee-dohm observed as well. This is from the end of his issue description (emphasis mine):

> Some things that got Atom to clear the diff info (**but not reliably**):

> **Keyboard tabbing to Slack and back**
> Closing the dev tools pane

**Comment by nathansobo** [comment]
I've been able to reproduce the more sinister issue by splitting the pane after saving but before committing. Will fix that before tomorrow.

**Comment by nathansobo** [comment]
Fixed via 192997c8cf4479c4cae155207ad3b175c0ee9498.

**Comment by nathansobo** [comment]
So there was a related, much worse issue related to gutter misbehavior after splitting panes that's now fixed.

**Comment by benogle** [comment]
Did you mean to reopen?

**Comment by nathansobo** [comment]
@benogle Yeah, this is actually still a problem. I fixed something else. This has probably been a problem for a while.

**Comment by nathansobo** [comment]
Okay, after some indecision, deciding to close this issue. I fixed the behavior described in the **title** of the issue. But it was caused by a different set of reproduction steps than those listed in the **body** of the issue. @lee-dohm maybe you want to open another issue relating to the diff hanging around until the focus changes? It's a different problem than the overlapping numbers etc described in the title.

So, once and for all: Closed via 192997c8cf4479c4cae155207ad3b175c0ee9498
